packages/blade/src/components/Checkbox/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -127,6 +128,8 @@ const CheckboxGroup = ({
           position={labelPosition}
           id={ids.labelId}
           accessibilityText={accessibilityText}
+          spacingRight={labelPosition === 'left' ? 'spacing.5' : 'spacing.0'}
+          spacingBottom={labelPosition === 'top' || isReactNative ? 'spacing.2' : 'spacing.0'}
```

Comment:
Explicitly adding margins for labels on individual components instead of relying on the Form label to add these margins. Better to keep the re-usable component free of margin styling so that individual components can place them with appropriate styles